securebanking-openbanking-uk-rs-simulator-server/src/main/java/com/forgerock/securebanking/openbanking/uk/rs/api/discovery/AvailableApiEndpointsResolver.java

```diff
@@ -80,6 +70,7 @@ public AvailableApiEndpointsResolver(RequestMappingHandlerMapping requestHandler
 
             if (apiReference != null) {
                 String version = getVersion(requestMapping);
+                String baseUrl = ServletUriComponentsBuilder.fromCurrentContextPath().build().toUriString();
```

Comment:
Fortunately Spring is clever enough to use the X-Forwarded headers if they're provided.